### PR TITLE
refactor(foreign-arb-with-appeals): upgrade to dispute resolver v2.0.0

### PR DESCRIPTION
**Pull request by ferittuncer**

### CODE REVIEW

**Review by hbarcelos — CHANGES_REQUESTED** [review]
image
There are some tests regarding removed functionality that are failing.
They should be removed as well.

**Comment by ferittuncer** [comment]
Fixed.

File: contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol [hunk 1]
```diff
@@ -621,25 +611,24 @@ contract RealitioForeignArbitrationProxyWithAppeals is IForeignArbitrationProxy,
...
+                    : 0;
+            } else if (finalAnswer == _contributedTo) {
+                uint256 paidFees = round.paidFees[_contributedTo];
+                // Reward the winner.
+                sum += paidFees > 0
+                    ? (round.contributions[_beneficiary][_contributedTo] * round.feeRewards) / paidFees
+                    : 0;
             }
         }
         return sum;
```
Comment (shalzz) [inline contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol:634]: `return sum` is unnecessary here.

File: contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol [hunk 2]
```diff
@@ -390,43 +396,23 @@ contract RealitioForeignArbitrationProxyWithAppeals is IForeignArbitrationProxy,
     }
 
     /**
-     * @notice Allows to withdraw any reimbursable fees or rewards after the dispute gets solved for multiple ruling options (answers) at once.
-     * @dev This function is O(n) where n is the number of queried answers.
-     * @dev This could exceed the gas limit, therefore this function should be used only as a utility and not be relied upon by other contracts.
+     * @notice Allows to withdraw any rewards or reimbursable fees for all rounds at once.
+     * @dev This function is O(n) where n is the total number of rounds. Arbitration cost of subsequent rounds is `A(n) = 2A(n-1) + 1`.
+     *      So because of this exponential growth of costs, you can assume n is less than 10 at all times.
```
Comment (unknownunknown1) [inline contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol:401]: There are extra spaces at the beginning

File: contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol [hunk 3]
```diff
@@ -390,43 +396,23 @@ contract RealitioForeignArbitrationProxyWithAppeals is IForeignArbitrationProxy,
...
     function withdrawFeesAndRewardsForAllRounds(
         uint256 _arbitrationID,
         address payable _beneficiary,
-        uint256[] memory _contributedTo
+        uint256 _contributedTo
     ) external override {
         address requester = arbitrationIDToRequester[_arbitrationID];
         ArbitrationRequest storage arbitration = arbitrationRequests[_arbitrationID][requester];
 
         for (uint256 roundNumber = 0; roundNumber < arbitration.rounds.length; roundNumber++) {
```
Comment (unknownunknown1) [inline contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol:414]: `arbitration.rounds.length` can be put into a local variable

**Review by unknownunknown1 — COMMENTED** [review]
A couple more points that aren't explicitely related to the changes, which mirror the latest changes of RealitioProxy:
1) We can make multipliers constant and remove governance altogether
2) Require in `submitEvidence` can be removed. We can only leave the event emission
3) In `fundAppeal` reimburse only when the remained is not 0

File: contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol (hunk 2 above)
Comment (ferittuncer) [reply contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol:401]: Yeah, I aligned it to above on purpose. Like a paragraph.

**Pull request merged**